components/widget/widget_util.lua

```diff
@@ -0,0 +1,30 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:Widget/Util
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Array = require('Module:Array')
+local Table = require('Module:Table')
+
+local Util = {}
+
+function Util.collect(...)
+	local elements = Table.pack(...)
+	local array = {}
+	for index = 1, elements.n do
+		local x = elements[index]
+		if Array.isArray(x) then
+			for _, y in ipairs(x) do
+				table.insert(array, y)
+			end
```

Comment:
possibly Array.extendWith?

Reply:
I rather control it explicitly, there's a reason why I went with this over any existing solution